frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import { IconColor } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useState } from '#libs/hooks/hooks.js';
+
+import { Button } from '../components.js';
+import {
+  INITIAL_STEP,
+  NEXT_STEP_INCREMENT,
+} from './libs/constants/constants.js';
+import styles from './styles.module.scss';
+
+type StepComponentProperties = {
+  onNext?: () => void;
+  onClose?: () => void;
+};
+
+type Step = {
+  component: React.ReactElement<StepComponentProperties>;
+  title: string;
+};
+
+type Properties = {
+  children?: React.ReactElement<StepComponentProperties>;
+  steps?: Step[];
+  isDisplayed: boolean;
+  title?: string;
+  onClose?: () => void;
+};
+
+const Modal: React.FC<Properties> = ({
+  children,
+  steps,
+  isDisplayed,
+  title,
+  onClose,
+}) => {
+  const [currentStep, setCurrentStep] = useState(INITIAL_STEP);
+
+  const goToNextStep = (): void => {
+    setCurrentStep(currentStep + NEXT_STEP_INCREMENT);
+  };
+
+  const handleClose = useCallback((): void => {
+    setCurrentStep(INITIAL_STEP);
+    onClose?.();
+  }, [onClose]);
+
+  const isMultiStep = Array.isArray(steps) && steps.length > INITIAL_STEP;
+
+  const currentComponent = isMultiStep
+    ? steps[currentStep]?.component
+    : children;
+  const currentTitle = isMultiStep ? steps[currentStep]?.title : title;
+
+  const renderedComponent = React.isValidElement(currentComponent)
+    ? React.cloneElement(currentComponent, {
+        onNext: goToNextStep,
+        onClose: handleClose,
+      })
+    : null;
+
+  return (
+    <dialog
+      open={isDisplayed}
```

Comment:
Check how we are handling references at now [here](frontend/src/libs/components/modal/modal.tsx) as @v0ldemar01 suggested, because we have to synchronize our effortsю The same approach should be used in your modal wrapper